common/mails/courses/index.ts

```diff
@@ -183,3 +184,39 @@ export async function sendParticipantCourseCertificate(participant: Pupil, cours
     //send mail 🎉
     await sendTemplateMail(mail, participant.email);
 }
+
+export async function sendPupilCourseSuggestion(course: Course | Prisma.course, subcourse: Subcourse | Prisma.subcourse) {
+    const minGrade = subcourse.minGrade;
+    const maxGrade = subcourse.maxGrade;
+
+    let lectures = await prisma.lecture.findMany({ where: { subcourseId: subcourse.id } });
+    let firstLecture = lectures[0].start;
+
+    for (let i = 1; i < lectures.length; i++) {
+        if (lectures[i].start < firstLecture) {
+            firstLecture = lectures[i].start;
+        }
+    }
+    const grades = [];
+
+    for (let grade = minGrade; grade <= maxGrade; grade++) {
+        grades.push(`${grade}. Klasse`);
+    }
+
+    const pupils = await prisma.pupil.findMany({
```

Comment:
This will potentially notify about 10.000 users. Please clarify with the stakeholders whether we really want to notify all users, or just the ones who had recent courses or where active recently or something like that.